source/autoSettingsUtils/driverSetting.py

```diff
@@ -21,130 +23,72 @@ class DriverSetting(AutoPropertyObject):
 	Used for synthesizer or braille display setting such as voice, variant or dot firmness as
 	well as for settings in Vision Providers
 	"""
+	#: Internal identifier of the setting.
 	id: str
-	displayName: str
+	#: The localized string shown in voice or braille settings dialog.
 	displayNameWithAccelerator: str
-	availableInSettingsRing: bool
-	defaultVal: object
-	useConfig: bool
-
-	#: Type information for _get_configSpec
-	configSpec: str

```

Comment:
The dynamic nature of `AutoPropertyObject` means that (at least some) tools don't recognize that `_get_configSpec` creates a property. Is there a way we can still keep this, or is there an alternative approach to making `_get_configSpec` a property?

Reply:
Are you intending to ask whether we can avoid AutoPropertyObject for this class?

Reply:
Sorry, by "this" I meant `configSpec: str` which tells tools that the property exists and it's type. I either want to keep the type information, so that tools know that this property exists (and what its type is) or mark `configSpec` as a property some other way. I'm really saying that, despite the fact that AuoPropertyObject turns this into a property, it's not recognized by tools. Given the syntactic sugar available in python now, I think it might soon be time to have look at what it would take to remove `AutoPropertyObject`.

Reply:
I think we should be able to patch AutoPropertyObject in such a way that it adds annotations. However, I wonder whether this would have any benefit for type checking in IDEs. I will write a prototype, if you could test that, that would be most helpful.

Reply:
@feerrenrut: could you have a look at https://github.com/leonardder/nvda/tree/autopropertyTypeChecking ?

Reply:
I actually think this isn't going to work, as I assume every linter doesn't create the classes, and as the annotations are added on the meta class, it will only work at runtime inspection of the class. That's a pitty. I will investigate another idea I have.

Reply:
> it will only work at runtime inspection of the class. 

Yes, I just had a look at this, that would be my concern as well.

Reply:
I've found a proper workaround for this.